packages/sanity/src/core/store/_legacy/document/document-pair/validation.ts

```diff
@@ -36,9 +36,17 @@ export const validation = memoize(
     },
     {draftId, publishedId, versionId}: IdPair,
     typeName: string,
+    displayedDocumentId?: string,
   ): Observable<ValidationStatus> => {
     const document$ = editState(ctx, {draftId, publishedId, versionId}, typeName).pipe(
-      map(({version, draft, published}) => version || draft || published),
+      map((state) => {
+        const {version, draft, published} = state
+
+        if (displayedDocumentId) {
+          return [version, draft, published].find((doc) => doc?._id === displayedDocumentId)
```

Comment:
Just to double check for peace of mind, is this order correct? I'm just asking since in quite a few other places the order tends to be the inverse (published first). Or at least I have this vague memory of it, just wanted to double check 🙏